view/templates/field_input.tpl

```diff
@@ -1,7 +1,7 @@
 	
 	<div class="field input" id="wrapper_{{$field.0}}">
 		<label for="id_{{$field.0}}">{{$field.1}}</label>
-		<input{{if $field.6 eq 'email'}} type="email"{{elseif $field.6 eq 'url'}} type="url"{{elseif $field.6 eq 'int'}} type="number"{{else}} type="text"{{/if}} name="{{$field.0}}" id="id_{{$field.0}}" value="{{$field.2 nofilter}}"{{if $field.4 eq 'required'}} required{{/if}}{{if $field.5 eq "autofocus"}} autofocus{{elseif $field.5}} {{$field.5 nofilter}}{{/if}} aria-describedby="{{$field.0}}_tip">
+		<input{{if $field.6}} type="{{$field.6}}"{{else}} type="text"{{/if}} name="{{$field.0}}" id="id_{{$field.0}}" value="{{$field.2 nofilter}}"{{if $field.4 eq 'required'}} required{{/if}}{{if $field.5 eq "autofocus"}} autofocus{{elseif $field.5}} {{$field.5 nofilter}}{{/if}} aria-describedby="{{$field.0}}_tip">
```

Comment:
Suggestion: `type="{{$field.6 ?? 'text'}}"`

Reply:
that works? wow .. I'm more and more learning ^^ .. I now know why I normally better stay at the backend ^^

Reply:
Please test first, Smarty template allow a bunch of PHP syntax, which makes me believe we would be better off using pure PHP templates.

Reply:
my local instance cries:
```console
replaceMacros: Syntax error in template "file:/vagrant/view/templates/field_input.tpl"  on line 4 ""  - Unexpected " ?", expected one of: "}}"
```

Reply:
I know from Level-2 (who invented Dice) that they introduced a "sample like" template engine. So you create a plain HTML doc with sample-text and than the engine replaces the sample-text during rendering. It's mainly build for frontend-engineers so they can create new pages without the need to know about PHP-internal stuff ;-) .. https://github.com/Level-2/Transphporm

Reply:
Maybe `$field.6 ? $field.6 : 'text'` ?

Reply:
> It's mainly build for frontend-engineers so they can create new pages without the need to know about PHP-internal stuff

I still don't believe these exist. It's been the justification for every single templating engine and I've never seen anyone use a templating engine without also doing the underlying PHP backend which is super-dumb.

Reply:
> > It's mainly build for frontend-engineers so they can create new pages without the need to know about PHP-internal stuff
> 
> I still don't believe these exist. It's been the justification for every single templating engine and I've never seen anyone use a templating engine without also doing the underlying PHP backend which is super-dumb.

just have a look at the link :) At least they drop the whole variable-stuff out of the plain templates and overwrite the values at the CSS base.

Reply:
> Maybe `$field.6 ? $field.6 : 'text'` ?

computer says no :-) I let it as is to not brick something..

Reply:
> just have a look at the link :) At least they drop the whole variable-stuff out of the plain templates and overwrite the values at the CSS base.

I did, and now I want to gouge my eyes out. It basically moves code logic from templates into having presentation logic in the code, at the price of having an additional intermediate template file, which isn't ideal either.

Reply:
At least it fully seperate the design from the template engine itself ;) but it just came across my way during some search

Reply:
This is true, but you need the design mock-up to be actually functional for this to be true. And if the design mock-up changes, the TSS files have to be updated, with a high potential for dead selectors as with CSS... Ugh Ugh Ugh.